lib/cocoapods-core/specification.rb

```diff
@@ -125,8 +125,9 @@ def hash
     #         clients.
     #
     def to_s
-      if name && !version.version.empty?
-        "#{name} (#{version})"
+      specified_version = attributes_hash['version'] || ''
+      if name && !specified_version.empty?
+        "#{name} (#{specified_version})"
```

Comment:
needed because the validator calls `to_s` on specs, and this would crash on any invalid version string